SimG4CMS/Calo/src/CaloSD.cc

```diff
@@ -720,22 +720,24 @@ void CaloSD::update(const ::EndOfEvent*) {
   slave.get()->ReserveMemory(theHC->entries());
 
   int count(0);
-  int wrong(0);
   double eEM(0.0);
   double eHAD(0.0);
   double eEM2(0.0);
   double eHAD2(0.0);
 #ifdef EDM_ML_DEBUG
+  int wrong(0);
   double tt(0.0);
   double zloc(0.0);
   double zglob(0.0);
   double ee(0.0);
 #endif
   int hc_entries = theHC->entries();
   for (int i = 0; i < hc_entries; ++i) {
+#ifdef EDM_ML_DEBUG
     if (!saveHit((*theHC)[i])) {
```

Comment:
@aandvalenzuela , I think this is wrong. we should call `saveHit((*theHC)[i]` . better to move the `#ifdef EDM_ML_DEBUG .... #endif` inside the condition

Reply:
or if we want to avoid the condition then you can do
```
#ifdef EDM_ML_DEBUG
if (!saveHit((*theHC)[i])) {
  ++wrong;
}
#else
  saveHit((*theHC)[i])
#endif
```

may be there is better way to do it :-)

Reply:
Thanks!